src/sentry/tasks/unmerge.py

```diff
@@ -0,0 +1,468 @@
+    GroupTagKey.objects.filter(
+        group_id=group_id,
+    ).delete()
+
+    GroupTagValue.objects.filter(
+        group_id=group_id,
+    ).delete()
+
+    GroupRelease.objects.filter(
+        group_id=group_id,
+    ).delete()
+
+    tsdb.delete([
+        tsdb.models.group,
+    ], [group_id])
+
+    tsdb.delete_distinct_counts([
+        tsdb.models.users_affected_by_group,
+    ], [group_id])
+
+    tsdb.delete_frequencies([
+        tsdb.models.frequent_releases_by_group,
+        tsdb.models.frequent_environments_by_group,
+    ], [group_id])
+
+
+def collect_tag_data(events):
+    results = {}
+
+    for event in events:
+        tags = results.setdefault(event.group_id, {})
+
+        for key, value in event.get_tags():
+            values = tags.setdefault(key, {})
+
+            if value in values:
+                times_seen, first_seen, last_seen = values[value]
+                values[value] = (times_seen + 1, event.datetime, last_seen)
+            else:
+                values[value] = (1, event.datetime, event.datetime)
+
+    return results
+
+
+def repair_tag_data(project, events):
+    # Repair `GroupTag{Key,Value}` data.
+    for group_id, keys in collect_tag_data(events).items():
+        for key, values in keys.items():
+            GroupTagKey.objects.get_or_create(
+                project_id=project.id,
+                group_id=group_id,
+                key=key,
+            )
+
+            # XXX: `{first,last}_seen` columns don't totally replicate the
+            # ingestion logic (but actually represent a more accurate value.)
+            # See GH-5289 for more details.
+            for value, (times_seen, first_seen, last_seen) in values.items():
+                instance, created = GroupTagValue.objects.get_or_create(
+                    project_id=project.id,
+                    group_id=group_id,
+                    key=key,
+                    value=value,
+                    defaults={
+                        'first_seen': first_seen,
+                        'last_seen': last_seen,
+                        'times_seen': times_seen,
+                    },
+                )
+
+                if not created:
+                    instance.update(
+                        first_seen=first_seen,
+                        times_seen=F('times_seen') + times_seen,
+                    )
+
+
+def collect_release_data(project, events):
+    results = {}
+
+    for event in events:
+        release = event.get_tag('sentry:release')
+
+        # TODO: Double check this!
+        if not release:
+            continue
+
+        # XXX: It's not really clear what the canonical source is for
+        # environment between the tag and the data attribute, but I'm going
+        # with data attribute for now. Right now it seems like they are
+        # intended to both be present and the same value, but I'm not really
+        # sure that has always been the case for existing values.
```

Comment:
This is another case where I'm not sure what the implications of assuming this data has existed forever are.